### ISSUE TITLE
Bugs in example "Single date Picker" as local build (Web)

### ISSUE THREAD

**Issue by Tesla-Renting** [web-ridge/react-native-paper-dates#91]
### Steps to reproduce
- `expo init`
- enter name for project
- `Choose a template: › blank (TypeScript)`
- `cd` to project
- `expo install @expo/vector-icons react-native-paper react-native-paper-dates`
- copy contents of the example to App.tsx
```
import * as React from 'react';
import { Button } from 'react-native-paper';
import { DatePickerModal } from 'react-native-paper-dates';

export default function ReadMeExampleSingle() {
  const [date, setDate] = React.useState<Date | undefined>(undefined);
  const [open, setOpen] = React.useState(false);

  const onDismissSingle = React.useCallback(() => {
    setOpen(false);
  }, [setOpen]);

  const onConfirmSingle = React.useCallback(
    (params) => {
      setOpen(false);
      setDate(params.date);
    },
    [setOpen, setDate]
  );

  return (
    <>
      <Button onPress={() => setOpen(true)} uppercase={false} mode="outlined">
        Pick single date
      </Button>
      <DatePickerModal
        // locale={'en'} optional, default: automatic
        mode="single"
        visible={open}
        onDismiss={onDismissSingle}
        date={date}
        onConfirm={onConfirmSingle}
        // validRange={{
        //   startDate: new Date(2021, 1, 2),  // optional
        //   endDate: new Date(), // optional
        // }}
        // onChange={} // same props as onConfirm but triggered without confirmed by user
        // saveLabel="Save" // optional
        // label="Select date" // optional
        // animationType="slide" // optional, default is 'slide' on ios/android and 'none' on web
      />
    </>
  );
}
```
- `expo start --web`

### Result
<img src="https://user-images.githubusercontent.com/85846148/127578139-5512271e-f3c4-4a24-b15e-f0d9026ef551.png" height="400"/>

- all elements of the `DatePickerModal` are collapsed to the position x=0 of the window
- the button `Pick single date` does not do anything

**Comment by mogarick** [comment]
@Tesla-Renting I faced the same problem but finally found the route cause is the  version of the `react-native-web` lib. this component requires versión `0.14.0` or greater.
Expo comes with versión `0.13.x` so you need to manually upgrade it via:

```
npm install react-native-web@0.14
```

The presuming root cause is this lib uses the `react-native` Modal component and it was implemented on React Native Web in version `0.14.0`.

**Comment by Tesla-Renting** [comment]
> @Tesla-Renting I faced the same problem but finally found the route cause is the version of the `react-native-web` lib. this component requires versión `0.14.0` or greater.
> Expo comes with versión `0.13.x` so you need to manually upgrade it via:
> 
> ```
> npm install react-native-web@0.14
> ```
> 
> The presuming root cause is this lib uses the `react-native` Modal component and it was implemented on React Native Web in version `0.14.0`.
> Also notice that you cannot install other versión of RNW. Versión `0.15` onwards needs React v17 and expo currently uses v16.
> 
> Regarding the button doing "nothing" is also related to the versión of RNW. It doesn't work on `0.13`. This is presumably related to the last version of `react-native-paper` that maybe also requires some functionality not present on v `0.13`.

Also `expo install react-native-web` updates the package to the latest version (currently 0.17.1). That solves the problem, but I am getting some errors and warnings in the console of the browser:
```
Warning: React does not recognize the `enterKeyHint` prop on a DOM element. If you intent[...]
```
```
Animated: `useNativeDriver` is not supported because the native animated module is missing. Falling ba[...]
```
but these are probably related to non-web-environments and are not correctly handled in the `react-native-web` version.

**Comment by mogarick** [comment]
> 
> Also `expo install react-native-web` updates the package to the latest version (currently 0.17.1). That solves the problem, but I am getting some errors and warnings in the console of the browser:
> 
> ```
> Warning: React does not recognize the `enterKeyHint` prop on a DOM element. If you intent[...]
> ```
> 
>  ```
> Animated: `useNativeDriver` is not supported because the native animated module is missing. Falling ba[...]
> ```
> 
> but these are probably related to non-web-environments and are not correctly handled in the `react-native-web` version.

It could also have to do with the fact that from version RNW 0.15 onwards React 17 is a peer dependency and maybe something else gets broken. I didn't update the Expo React version, so I stuck to React v16 and a so RNW 0.15+ is not supported formally.
